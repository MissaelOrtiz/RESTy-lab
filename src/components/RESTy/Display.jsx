import React from 'react';
import PropTypes from 'prop-types';
import ReactJson from 'react-json-view';

const Display = ({ res }) => (
  <div>
    <ReactJson src={res} />
  </div>
);

Display.propTypes = {
  res: PropTypes.object || PropTypes.array,
};

export default Display;
