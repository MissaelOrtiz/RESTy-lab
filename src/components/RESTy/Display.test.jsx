/* eslint-disable max-len */
import React from 'react';
import { render, screen } from '@testing-library/react';
import Display from './Display';

describe('Display', () => {
  it('displays a json object', () => {
    render(
      <Display 
        res={{ hello: 'world' }}
      />
    );

    const display = screen.getByText('"world"');
    expect(display).toMatchSnapshot();
  });
});
