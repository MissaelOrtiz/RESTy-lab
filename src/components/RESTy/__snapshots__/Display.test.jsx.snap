// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Display displays a json object 1`] = `
<span
  class="string-value"
  style="cursor: default;"
>
  "
  world
  "
</span>
`;
